### PR TITLE
Add primitive cli

### PR DESCRIPTION
**Pull request by kcudnik**

### CODE REVIEW

File: syncd/syncd_cli.cpp [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+    {
+        SWSS_LOG_ERROR("open socket failed, errno: %s", strerror(errno));
+        return;
+    }
+
+    struct sockaddr_in serv_addr;
+
+    memset(&serv_addr, 0, sizeof(serv_addr));
+
+    serv_addr.sin_family = AF_INET;
```
Comment (qiluo-msft) [inline syncd/syncd_cli.cpp]: >AF_INET [](start = 27, length = 7)

Add unix socket as an alternative?

File: syncd/syncd_cli.cpp [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+}
+
+void cmd_help(const std::vector<std::string>& args)
+{
+    SWSS_LOG_ENTER();
+
+    std::string help = "\n\
+    help            - this command\n\
+    exit            - close cli connection\n\
+    loglevel LEVEL  - sets debug logger level\n\n";
```
Comment (qiluo-msft) [inline syncd/syncd_cli.cpp]: >; [](start = 50, length = 1)

loglevel  - gets debug logger level\n\n

File: syncd/syncd_cli.cpp [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+            {
+                SWSS_LOG_ERROR("read failed, errno: %s", strerror(errno));
+            }
+
+            return;
+        }
+
+        const std::string& s = std::string(buffer, n);
+
+        handle_cli_commands(s);
```
Comment (qiluo-msft) [inline syncd/syncd_cli.cpp]: >handle_cli_commands [](start = 8, length = 19)

Is it possible that s contains multiple line? Considering redirect input from a file or pipe.

File: syncd/syncd_cli.cpp [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+
+        if (cli_client_socket < 0)
+        {
+            // don't log when run is false, that's mean we closed socket
+            if (run_cli)
+            {
+                SWSS_LOG_ERROR("accept failed, errno: %s", strerror(errno));
+            }
+
+            continue;
```
Comment (qiluo-msft) [inline syncd/syncd_cli.cpp]: >continue [](start = 12, length = 8)

continue->break?

File: syncd/syncd_cli.cpp [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+{
+    SWSS_LOG_ENTER();
+
+    char buffer[BUFFER_SIZE];
+
+    run_user = true;
+
+    while (run_user)
+    {
+        if (!sendtoclient(CLI_PROMPT))
```
Comment (qiluo-msft) [inline syncd/syncd_cli.cpp]: >sendtoclient(CLI_PROMPT [](start = 13, length = 23)

Don't send if stdin redirected.

File: syncd/syncd_cli.cpp (hunk 3 above)
Comment (qiluo-msft) [inline syncd/syncd_cli.cpp]: >handle_cli_commands [](start = 8, length = 19)

Is it possible that s contains truncated line? Also considering redirect input from a file or pipe, or user copy and paste a long line.

File: syncd/syncd_cli.cpp (hunk 2 above)
Comment (kcudnik) [inline syncd/syncd_cli.cpp]: its set not show

File: syncd/syncd_cli.cpp (hunk 3 above)
Comment (kcudnik) [inline syncd/syncd_cli.cpp]: yes, as pr stays, primitive

Comment (kcudnik) [inline syncd/syncd_cli.cpp]: yes it may not work in this case, lets not add everything in 1 pr

File: syncd/syncd_cli.cpp (hunk 2 above)
Comment (kcudnik) [inline syncd/syncd_cli.cpp]: feature request, you are wellcome to add it :)

File: syncd/syncd_cli.cpp [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+        {
+            return;
+        }
+
+        // blocking
+        ssize_t n = read(cli_client_socket, buffer, BUFFER_SIZE - 1);
```
Comment (kcudnik) [inline syncd/syncd_cli.cpp]: fixed

File: syncd/syncd_cli.cpp (hunk 4 above)
Comment (kcudnik) [inline syncd/syncd_cli.cpp]: fixed

**Pull request merged**